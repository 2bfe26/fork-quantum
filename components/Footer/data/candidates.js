export default {
  id: 1,
  title: 'Candidatos',
  cols: 8,
  theme: {
    primaryColor: '#005da4',
    secondaryColor: '#666666',
    titleFontSize: '14px',
    titleFontWeight: 'bold',
    linkFontSize: '12px',
  },
  linkGroups: [{
    id: 1,
    sideSeparator: true,
    cols: 4,
    links: [{
      id: 1,
      title: true,
      text: 'Buscar vagas',
      href: null,
      target: null,
      img: null,
    }, {
      id: 2,
      title: false,
      text: 'Por palavra-chave',
      href: '/vagas/',
      target: '_self',
      img: null,
    }, {
      id: 3,
      title: false,
      text: 'Por área/cargo',
      href: 'https://www.catho.com.br/vagas/por-area/',
      target: '_self',
      img: null,
    }, {
      id: 4,
      title: false,
      text: 'Melhores e maiores empresas do Brasil',
      href: 'https://www3.catho.com.br/vag/empregos-maiores-empresas/',
      target: '_self',
      img: null,
    }],
  }, {
    id: 2,
    sideSeparator: true,
    cols: 4,
    links: [{
      id: 1,
      title: true,
      text: 'Produtos Gratuitos',
      href: null,
      target: null,
      img: null,
    }, {
      id: 2,
      title: false,
      text: 'Aviso de vagas e buscas salvas',
      href: '/aviso-vagas/listagem/',
      target: '_self',
      img: null,
    }, {
      id: 3,
      title: false,
      text: 'Buscar vagas',
      href: 'https://www.catho.com.br/vagas/',
      target: '_self',
      img: null,
    }, {
      id: 4,
      title: false,
      text: 'Guia de Profissões e Salários',
      href: '/profissoes/',
      target: '_self',
      img: null,
    }, {
      id: 5,
      title: false,
      text: 'Por dentro das empresas',
      href: '/por-dentro/',
      target: '_self',
      img: null,
    }],
  }, {
    id: 3,
    sideSeparator: false,
    cols: 4,
    links: [{
      id: 1,
      title: true,
      text: 'Apoio ao candidato',
      href: null,
      target: null,
      img: null,
    }, {
      id: 2,
      title: false,
      text: 'Elaboração de Currículo',
      href: '/https://parceiros.catho.com.br/gop.php?origem=cat&a=6515&p=12&url1=1&url=servicoscatho%2Fcarreira%2Felaboracao-de-curriculo%2F',
      target: '_self',
      img: null,
    }, {
      id: 3,
      title: false,
      text: 'Análise de Currículo',
      href: 'https://parceiros.catho.com.br/gop.php?origem=cat&a=6523&p=4&url1=1&url=servicoscatho%2Fcarreira%2Fanalise-de-curriculo%2F',
      target: '_self',
      img: null,
    }, {
      id: 4,
      title: false,
      text: 'Simulação de Entrevista',
      href: 'https://parceiros.catho.com.br/gop.php?origem=cat&a=6527&p=16&url1=1&url=servicoscatho%2Fcarreira%2Fsimulacao-de-entrevista%2F',
      target: '_self',
      img: null,
    }, {
      id: 5,
      title: false,
      text: 'Elaboração de Carta de Apresentação',
      href: 'https://parceiros.catho.com.br/gop.php?origem=cat&a=6519&p=13&url1=1&url=servicoscatho%2Fcarreira%2Fcarta-de-apresentacao%2F',
      target: '_self',
      img: null,
    }, {
      id: 6,
      title: false,
      text: 'Conheça Todas as Soluções',
      href: '/servicoscandidatos/',
      target: '_self',
      img: null,
    }],
  }, {
    id: 4,
    sideSeparator: true,
    cols: 4,
    links: [{
      id: 1,
      title: true,
      text: 'Carreira e sucesso',
      href: 'https://www.catho.com.br/carreira-sucesso/',
      target: '_self',
      img: null,
    }],
  }, {
    id: 5,
    sideSeparator: true,
    cols: 4,
    links: [{
      id: 1,
      title: true,
      text: 'Exclusivo para assinantes',
      href: null,
      target: null,
      img: null,
    }, {
      id: 2,
      title: false,
      text: 'Cursos Rápidos',
      href: '/cursos-rapidos/',
      target: '_self',
      img: null,
    }, {
      id: 3,
      title: false,
      text: 'Cadastro Catho',
      href: 'https://cadastro.catho.com.br/',
      target: '_self',
      img: null,
    }, {
      id: 4,
      title: false,
      text: 'Consultor Virtual',
      href: '/consvirtual/',
      target: '_self',
      img: null,
    }],
  }, {
    id: 6,
    sideSeparator: false,
    cols: 4,
    links: [{
      id: 1,
      title: true,
      text: 'Ajuda',
      href: null,
      target: null,
      img: null,
    }, {
      id: 2,
      title: false,
      text: 'Central de Relacionamento',
      href: 'https://www.catho.com.br/sac/',
      target: '_self',
      img: null,
    }, {
      id: 3,
      title: true,
      text: 'Promoção 7 dias grátis',
      href: null,
      target: null,
      img: null,
    }, {
      id: 5,
      title: false,
      text: 'Entenda a promoção',
      href: 'https://www.catho.com.br/help/usr/faq_promocao.phtml?home=1&width=650px&height=600px&iframe=true',
      target: '_self',
      img: null,
    }],
  }],
};
