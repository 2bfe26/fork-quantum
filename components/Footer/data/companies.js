export default {
  id: 1,
  title: 'Empresas',
  cols: 2,
  theme: {
    primaryColor: '#005C45',
    secondaryColor: '#666666',
    titleFontSize: '14px',
    titleFontWeight: 'bold',
    linkFontSize: '12px',
  },
  linkGroups: [{
    id: 1,
    sideSeparator: false,
    cols: 12,
    links: [{
      id: 1,
      title: false,
      text: 'Solução Atração dos Melhores',
      href: 'https://seguro.catho.com.br/inc/novaInclusao/FormPrincipal.php?tipoProd=Vag&c=13006',
      target: '_self',
      img: null,
    }, {
      id: 2,
      title: false,
      text: 'Solução Recrutamento Perfeito',
      href: 'https://www.catho.com.br/empresas/produtos/',
      target: '_self',
      img: null,
    }, {
      id: 3,
      title: false,
      text: 'Produtos para Empresa',
      href: 'https://www.catho.com.br/empresas/produtos/',
      target: '_self',
      img: null,
    }],
  }],
};
