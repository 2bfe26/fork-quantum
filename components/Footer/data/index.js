import Candidates from './candidates';
import Companies from './companies';
import Courses from './courses';
import Institutional from './institutional';
import Stamps from './stamps';

const Data = {
  theme: {
    primaryColor: '#005da4',
    secondaryColor: '#666666',
    borderColor: '#e3e3e3',
    backgroundColor: '#f5f5f5',
    fontFamily: 'Arial',
    fontSize: '12px',
    paddingTop: '15px',
    paddingBottom: '10px',
    StampsHolderPadding: '10px',
  },
  candidates: Candidates,
  companies: Companies,
  courses: Courses,
  institutional: Institutional,
  stamps: Stamps,
};

export default Data;
