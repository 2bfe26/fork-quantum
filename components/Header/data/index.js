import Segments from './segments';
import Promotion from './promotion';
import Menu from './menu';

const Data = {
  theme: {
    primaryColor: '#005da4',
    secondaryColor: '#666666',
    borderColor: '#e3e3e3',
    backgroundColor: '#f5f5f5',
    fontFamily: 'Arial',
    fontSize: '12px',
    paddingTop: '10px',
    paddingBottom: '10px',
    StampsHolderPadding: '10px',
  },
  segments: Segments,
  promotion: Promotion,
  menu: Menu,
};

export default Data;
