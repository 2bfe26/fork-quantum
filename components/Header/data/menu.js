const common = [
  {
    id: 3,
    title: false,
    text: 'Aviso de vagas',
    href: '/aviso-vagas/listagem/',
    target: '_self',
    img: null,
  }, {
    id: 4,
    title: false,
    text: 'Envie seu Currículo por 7 dias grátis',
    href: 'https://seguro.catho.com.br/inclusao/curriculo.php?nova_inclusao=1&usr[plano_id]=1&cur[perfil_id]=1&cur[use_personalizado]=0&a=0&ppd=0&tipoinclusao=807',
    target: '_self',
    img: null,
  }, {
    id: 5,
    title: false,
    text: 'Guia de profissões',
    href: '/profissoes/',
    target: '_self',
    img: null,
  }, {
    id: 6,
    title: null,
    text: 'Por dentro das empresas',
    href: '/por-dentro/',
    target: '_self',
    img: null,
  }, {
    id: 7,
    title: null,
    text: 'Portal do currículo',
    href: 'https://www.catho.com.br/portaldocurriculo/',
    target: '_self',
    img: null,
  }, {
    id: 8,
    title: null,
    text: 'Vagas de estágio',
    href: 'https://www.catho.com.br/estagio/',
    target: '_self',
    img: null,
  }, {
    id: 9,
    title: 'Exclusivo para assinantes',
    text: null,
    href: null,
    target: null,
    img: null,
  }, {
    id: 10,
    title: null,
    text: 'Cursos rápidos',
    href: '/cursos-rapidos/',
    target: '_self',
    img: null,
  }, {
    id: 11,
    title: null,
    text: 'Cadastro Catho',
    href: 'https://cadastro.catho.com.br/',
    target: '_self',
    img: null,
  }, {
    id: 12,
    title: null,
    text: 'Consultor virtual',
    href: '/consvirtual/',
    target: '_self',
    img: null,
  }, {
    id: 13,
    title: null,
    text: 'Vantagens de assinar a Catho',
    href: '/cadastrar-curriculo/',
    target: '_self',
    img: null,
  }, {
    id: 14,
    title: 'Apoio ao candidato',
    text: null,
    href: null,
    target: null,
    img: null,
  }, {
    id: 15,
    title: null,
    text: 'Elaboração de currículo',
    href: 'https://parceiros.catho.com.br/gop.php?origem=cat&a=6503&p=12&url1=1&url=servicoscatho/carreira/elaboracao-de-curriculo/',
    target: '_self',
    img: null,
  }, {
    id: 16,
    title: null,
    text: 'Análise de currículo',
    href: 'https://parceiros.catho.com.br/gop.php?origem=cat&a=6507&p=4&url1=1&url=servicoscatho/carreira/analise-de-curriculo/',
    target: '_self',
    img: null,
  }, {
    id: 17,
    title: null,
    text: 'Simulação de entrevista',
    href: 'https://parceiros.catho.com.br/gop.php?origem=cat&a=6509&p=16&url1=1&url=servicoscatho/carreira/simulacao-de-entrevista/',
    target: '_self',
    img: null,
  }, {
    id: 18,
    title: null,
    text: 'Elaboração de carta de apresentação',
    href: 'https://parceiros.catho.com.br/gop.php?origem=cat&a=6505&p=13&url1=1&url=servicoscatho/carreira/carta-de-apresentacao/',
    target: '_self',
    img: null,
  }, {
    id: 19,
    title: null,
    text: 'Cursos',
    href: '/curso/',
    target: '_self',
    img: null,
  }, {
    id: 20,
    title: null,
    text: 'Todas as soluções',
    href: '/servicoscandidatos/',
    target: '_self',
    img: null,
  },
];

const desktop = {
  theme: {
    primaryColor: '#666666',
    secondaryColor: '#CCCCCC',
    tertiaryColor: '#333333',
    backgroundColor: '#FFFFFF',
    titleFontSize: '14px',
    titleFontWeight: 'bold',
    linkFontSize: '14px',
    marginLogo: '0px 20px 0px 0px',
    menuHover: {
      borderColor: '#005da4',
      borderSize: '2px',
    },
  },
  links: [{
    id: 1,
    title: false,
    text: 'Buscar vagas',
    href: '/vagas/',
    target: '_self',
    img: null,
  }, {
    id: 2,
    title: false,
    text: 'Guia de profissões',
    href: '/profissoes/',
    target: '_self',
    img: null,
  }, {
    id: 3,
    title: null,
    text: 'Por dentro das empresas',
    href: '/por-dentro/',
    target: '_self',
    img: null,
  }, {
    id: 4,
    title: false,
    text: 'Soluções para candidatos',
    href: null,
    target: null,
    img: null,
    collapsedLinkGroup: common,
  }],
};

const phone = {
  theme: {
    primaryColor: '#666666',
    secondaryColor: '#CCCCCC',
    tertiaryColor: '#333333',
    titleFontSize: '14px',
    titleFontWeight: 'bold',
    linkFontSize: '12px',
  },
  links: [{
    id: 1,
    title: false,
    text: 'Buscar vagas',
    href: '/vagas/',
    target: '_self',
    img: null,
    icon: 'search',
  }, {
    id: 2,
    title: null,
    text: 'Por dentro das empresas',
    href: '/por-dentro/',
    target: '_self',
    img: null,
    icon: 'enterprise',
  }, {
    id: 3,
    title: false,
    text: 'Aviso de vagas',
    href: '/aviso-vagas/listagem/',
    target: '_self',
    img: null,
    icon: 'alert',
  },
  ...common.slice(1)],
};

export default {
  desktop,
  phone,
};
