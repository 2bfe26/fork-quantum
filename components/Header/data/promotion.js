export default {
  theme: {
    primaryColor: '#da1d5d',
    secondaryColor: '#b6154e',
    textColor: '#fff',
  },
};
