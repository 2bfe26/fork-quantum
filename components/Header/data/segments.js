export default {
  id: 1,
  title: null,
  cols: 12,
  theme: {
    primaryColor: '#555',
    secondaryColor: '#CCCCCC',
    tertiaryColor: '#333333',
    titleFontSize: '14px',
    titleFontWeight: 'bold',
    linkFontSize: '12px',
  },
  links: [{
    id: 1,
    title: false,
    text: 'candidatos',
    href: '/',
    target: '_self',
    img: null,
  }, {
    id: 2,
    title: false,
    text: 'empresas',
    href: 'https://www.catho.com.br/empresa/',
    target: '_self',
    img: null,
  }, {
    id: 3,
    title: false,
    text: 'cursos',
    href: 'https://www.catho.com.br/educacao/',
    target: '_self',
    img: null,
  }, {
    id: 4,
    title: false,
    text: null,
    href: 'https://www.catho.com.br/help/',
    target: '_self',
    img: {
      src: 'header-info.svg',
      width: '16px',
      height: '16px',
      alt: '',
    },
  }],
};
