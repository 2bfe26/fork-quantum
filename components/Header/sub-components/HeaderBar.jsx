import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import { Container, BREAKPOINTS } from '../../Grid';

const HeaderContainer = styled(Container)`
  background-color: ${props => props.theme.backgroundColor};
  padding-top: ${props => props.theme.paddingTop};
  padding-bottom: ${props => props.theme.paddingBottom};

  @media (max-width: 980px) {
    background-color: ${props => props.theme.tertiaryColor};
    color: ${props => props.theme.secondaryColor};
  }
`;

const StyledContainer = styled(Container)`
  padding: 0;
`;

const LinkList = styled.ul`
  font-size: ${props => props.theme.fontSize};
  display: flex;
  align-items: center;
  padding: 0;
  margin: 0;

  @media (max-width: ${BREAKPOINTS.tablet}px) {
    color: ${props => props.theme.secondaryColor};
  }
`;

const LinkListItem = styled.li`
  color: ${props => props.theme.secondaryColor};
  margin-right: 25px;

  &:first-child, &:last-child {
    list-style-type: none;
  }

  &:last-child {
    margin-left: auto;
    margin-right: 0;
  }
`;

const Link = styled.a`
  color: ${props => props.theme.primaryColor};
  font-size: ${props => props.theme.linkFontSize};
  font-family: ${props => props.theme.fontFamily};
  text-decoration: none;

  &:hover{
    text-decoration: underline;
  }

  &.selected {
    font-weight: bold;
    cursor: default;
  }

  display: flex;

  img{
    margin-top:-2px;
    margin-left:6px;
  }

  @media (max-width: ${BREAKPOINTS.desktop}px) {
    color: ${props => props.theme.secondaryColor};
`;

const HeaderBar = ({ data, path }) => (
  <ThemeProvider theme={data.theme}>
    <HeaderContainer fluid>
      <StyledContainer>
        <LinkList>
          {
            data.links.map(link => (
              <LinkListItem key={link.id}>
                <Link href={link.href} className={path === link.href ? 'selected' : ''}>
                  { link.text }
                  {
                    (link.img) &&
                    <img
                      alt={link.text}
                      src={`${process.env.APPLICATION_PATH}/static/${link.img.src}`}
                      width={link.img.width}
                      height={link.img.height}
                    />
                  }
                </Link>
              </LinkListItem>
            ))
          }
        </LinkList>
      </StyledContainer>
    </HeaderContainer>
  </ThemeProvider>
);

HeaderBar.propTypes = { data: PropTypes.instanceOf(Object), path: PropTypes.string.isRequired };
HeaderBar.defaultProps = { data: null };

export default HeaderBar;
