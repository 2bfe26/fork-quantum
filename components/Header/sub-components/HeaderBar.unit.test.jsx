import React from 'react';
import renderer from 'react-test-renderer';
import HeaderBar from './HeaderBar';
import data from '../data';

describe('HeaderBar component', () => {
  it('Should match the snapshot', () => {
    const tree = renderer.create(<HeaderBar data={data.segments} path="/" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
