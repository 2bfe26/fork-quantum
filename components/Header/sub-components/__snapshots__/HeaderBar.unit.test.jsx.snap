// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar component Should match the snapshot 1`] = `
<div
  className="sc-ifAKCX fKVinI sc-bdVaJa OkJiS"
>
  <div
    className="sc-EHOje eHQBuR sc-bdVaJa cSQWIE"
  >
    <ul
      className="sc-bZQynM cXedio"
    >
      <li
        className="sc-gzVnrw dHPSlY"
      >
        <a
          className="selected sc-htoDjs bhCKaH"
          href="/"
        >
          candidatos
        </a>
      </li>
      <li
        className="sc-gzVnrw dHPSlY"
      >
        <a
          className="sc-htoDjs bhCKaH"
          href="https://www.catho.com.br/empresa/"
        >
          empresas
        </a>
      </li>
      <li
        className="sc-gzVnrw dHPSlY"
      >
        <a
          className="sc-htoDjs bhCKaH"
          href="https://www.catho.com.br/educacao/"
        >
          cursos
        </a>
      </li>
      <li
        className="sc-gzVnrw dHPSlY"
      >
        <a
          className="sc-htoDjs bhCKaH"
          href="https://www.catho.com.br/help/"
        >
          <img
            alt={null}
            height="16px"
            src="undefined/static/header-info.svg"
            width="16px"
          />
        </a>
      </li>
    </ul>
  </div>
</div>
`;
